### PR TITLE
ROX-21131: Add conditional rendering for no clusters page

### PR DESCRIPTION
**Pull request by pedrottimark**
## Description

A few more changes than I had intended because during manual testing, an unexpected GET /v1/init-bundles request directed attention at a loophole in complicated conditional rendering because
* `fetchingClusters` has initial value `false`
* `pollingCount` has initial value `0`
* `currentClusters.length` has initial value `0`

Which rendered NoClustersPage **before** first successful response.

### Updates to data flow

1. Visit or redirect to /main clusters which makes GET /v1/clusters request every 5 seconds.
2. Before first successful response, render either error message or spinner.
3. After first successful response:
    * If no clusters and not filtered, render NoClustersPage.
    * If clusters or filtered, render clusters table and so on.

### Changes for data flow

1. Replace `fetchingClusters` with opposite `hasFetchedClusters` to simplify conditional rendering of spinner during initial request. We might restore `isFilteringClusters` if the team develops a reusable component as recently discussed. However, it does not apply to ordinary polled requests.
2. Simplify and lift up conditional rendering with early return:
    * Before there is a response.
    * After there is a response, if there are no clusters nor search filter.
    * After there is a response, if there are clusters or search filter.

### Comments about data flow

1. It is important that /main/clusters NoClustersPage **Create bundle** goes to /main/clusters/init-bundles InitBundlesWizard in the same tab, so when **Download** goes back, NoClustersPage makes a new GET /v1/init-bundles request and therefore renders the link instead of the button.
2. It is important that /main/clusters NoClustersPage **Review installation instructions** opens /main/clusters/secure-a-cluster SecureCluster in new tab, so polling loop in original tab will cause conditional rendering of table whenever there is a secured cluster. That is, if it opens the page in the same tab, then it suggests the need for a back button outside of a wizard.

### Residue

1. After we merge this, move forward in parallel:
    * Add analytics.
    * Add integration tests.
2. As comments document, reconsider whether to factor out a minimal **Clusters** page heading so NoClustersPage (and therefore Spinner for initial fetch) has something in common with ordinary clusters page.
3. Factor out the clusters request and conditional rendering?
    * Rename ClustersPage as ClustersRoute like AdministrationEventsRoute.
    * Factor out the above as ClustersPage.

## Checklist
- [ ] Investigated and inspected CI test results
- [x] ~Unit test and regression tests added~

## Testing Performed

Before you `yarn deploy` in ui folder:

```sh
export ROX_MOVE_INIT_BUNDLES_UI=true
```

1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.css: 0 = 3950304 - 3950304
        total: 4298 = 10450113 - 10445815
    * `ls -al build/static/js/*.js | wc -l`
        0 = 90 - 90 files
3. `yarn start` in ui

### Manual testing

By the way, I noticed an inconsistency and added conditional rendering for **StackRox** in success alert if product branding is **StackRox** at upper left of page (see pictures).

1. Visit /main/clusters: see table
    StackRox_ClustersTablePanel

2. Visit /main/clusters with simulated no clusters nor init bundles.

    * See **Create bundle** button.
        StackRox_NoClustersPage_nor_bundles

    * Click to open init bundle wizard.

3. Click **Download** in wizard with simulated no clusters but init bundles.

    * See **Review installation instructions** link.
        StackRox_NoClustersPage_but_bundles

    * Click to open /main/clusters/secure-a-cluster page in a new tab so that the original tab displays clusters table whenever a cluster is secured.

### CODE REVIEW

File: ui/apps/platform/src/Containers/Clusters/ClustersTablePanel.tsx [hunk 1]
```diff
@@ -422,31 +454,38 @@ function ClustersTablePanel({
...
+                    {errorMessage && (
+                        <Alert
+                            variant="warning"
+                            isInline
+                            title="Unable to fetch clusters"
+                            component="p"
+                        >
+                            {errorMessage}
+                        </Alert>
+                    )}
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/Clusters/ClustersTablePanel.tsx:466]: Did you mean to have another possible error message here?

If so, did you mean for it to have the same title as the one above?

**Review by vjwilson — APPROVED** [review]
Just one question about one of the error messages.

File: ui/apps/platform/src/Containers/Clusters/ClustersTablePanel.tsx (hunk 1 above)
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Clusters/ClustersTablePanel.tsx:466]: Good eyes.
* Error message above is for **initial** request only (if I have reorganized state correctly).
* Error message here is for **subsequent** requests, so I decided to render it in addition to the table, which displays data from last successful request.

**Comment by pedrottimark** [comment]
ui-e2e-tests failures

> Clusters should redirect from Dashboard when no secured clusters have been added

Aha, I intended (but forgot) to make an assertion depend on feature flag for NoClustersPage versus AddCiusterPrompt.

**Pull request merged**